explorations/exploring_recipes.Rmd

```diff
@@ -0,0 +1,265 @@
+`hcai_missing` as a factor level.
+
+- A recipe is a exactly that. It's a process to transform data. Only the 
+process.
+- `prep` will gather the ingredients needed to apply the recipe. Example: One
+must know the mean before centering.
+- `bake` actually applies the recipe. Any subset of the same data can be 
+baked with the recipe.
+
+Basic examples can be found here:
+https://github.com/topepo/recipes/blob/master/vignettes/Simple_Example.Rmd
+
+Custom step examples can be found here:
+https://github.com/topepo/recipes/blob/master/vignettes/Custom_Steps.Rmd
+
+Or below for some starter code.
+
+```{r}
+library(caret)
+library(recipes)
+library(tidyverse)
+library(healthcareai)
+```
+
+
+Load a diabetes dataset. Download from `Box/Data Science Team/Classification/diabetes.csv`
+```{r}
+df <- as.tibble(read.csv("diabetic_data.csv", stringsAsFactors = FALSE))
+
+df$X <- NULL
+
+# Change to binary
+df$readmitted[df$readmitted==">30"] <- "N"
+df$readmitted <- as.factor(ifelse(df$readmitted == "<30", "Y", "N"))
+
+# Replace ? with NA
+df[df == "?"] <- NA
+
+cols_to_factor <- c("race", "gender", "admission_type_id", "discharge_disposition_id",
+  "admission_source_id", "payer_code", "medical_specialty", "diag_1", "diag_2", "diag_3", 
+  "metformin", "repaglinide", "nateglinide", "chlorpropamide", "glimepiride", 
+  "acetohexamide", "glipizide","glyburide","tolbutamide","pioglitazone","rosiglitazone",
+  "acarbose","miglitol","troglitazone","tolazamide","examide","citoglipton","insulin",
+  "glyburide.metformin","glipizide.metformin","glimepiride.pioglitazone","metformin.rosiglitazone",
+  "metformin.pioglitazone","change","diabetesMed", "readmitted")
+df[cols_to_factor] <- lapply(df[cols_to_factor], as.factor)
+
+df$age <- factor(df$age, ordered = TRUE)
+
+df[c("weight", "max_glu_serum", "A1Cresult")] <- lapply(
+  df[c("weight", "max_glu_serum", "A1Cresult")], as.numeric)
+
+data <- df %>% slice(1:20000)
+
+glimpse(data)
+
+print("size of dataset:")
+print(dim(data))
+
+print("unique patients:")
+print(length(unique(data$patient_nbr)))
+
+print("missing percentage:")
+print(countMissingData(data))
+
+print("structure:")
+print(str(data))
+```
+
+Train test split
+Splitting on the readmitted flag makes sure that the distribution is the same in both sets.
+```{r}
+train_index <- createDataPartition(data$readmitted,
+  p = 0.8,
+  times = 1,
+  list = TRUE)
+
+data_train <- data[train_index$Resample1, ]
+data_test <- data[-train_index$Resample1, ]
+```
+
+
+### Basic Recipe
+Use `?selections` to see how to select certain variables. One can select by type, role, or name
+This one does centering and scaling of numeric variables.
+```{r}
+# Initialize
+rec_obj <- recipe(readmitted ~ ., data = data)
+
+# Create recipe
+standardized <- rec_obj %>%
+  step_center(all_numeric()) %>%
+  step_scale(all_numeric()) 
+standardized
+
+# Train recipe
+trained_rec <- prep(standardized, training = data_train)
```

Comment:
I suspect you're being extra explicit here to show what's what, but in actual use, we'd just modify the same recipe / pipe it all the way through rather than create all these objects, right?

Reply:
Yes, definitely.